test/reduce.cpp

```diff
@@ -309,6 +309,525 @@ CPP_REDUCE_TESTS(anyTrue, any_true, float, unsigned char);
 CPP_REDUCE_TESTS(allTrue, all_true, float, unsigned char);
 CPP_REDUCE_TESTS(count, count, float, unsigned);
 
+//
+// Reduce By Key tests
+//
+
+// ReduceByKey helper methods
+//TODO: move to test data
+void getTest0_KeyVals(array &keys_out, array &vals_out) {
+    const int testSz = 8;
+    const int   h_keys[testSz] = { 0, 2, 2, 9, 5, 5, 5, 8 };
+    const float h_vals[testSz] = { 0, 7, 1, 6, 2, 5, 3, 4 };
+    array keys(testSz, h_keys);
+    array vals(testSz, h_vals);
+
+    keys_out = keys;
+    vals_out = vals;
+}
+
+void getTest0_Gold(array &keys_out, array &vals_out) {
```

Comment:
This is pretty much the same function as getTest0 but you have different values. This can be generalized.